src/m365/teams/commands/tab/tab-get.ts

```diff
@@ -0,0 +1,291 @@
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+import {
+  CommandOption, CommandValidate
+} from '../../../../Command';
+import GraphCommand from '../../../base/GraphCommand';
+import Utils from '../../../../Utils';
+import request from '../../../../request';
+import { Team } from '../../Team';
+import { Channel } from '../../Channel';
+import { Tab } from '../../Tab';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  teamId: string;
+  teamName: string;
+  channelId: string;
+  channelName: string;
+  tabId: string;
+  tabName: string;
+}
+
+class TeamsTabGetCommand extends GraphCommand {
+  public get name(): string {
+    return `${commands.TEAMS_TAB_GET}`;
+  }
+
+  public get description(): string {
+    return 'Gets information about the specified Microsoft Teams tab';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.teamId = args.options.teamId;
+    telemetryProps.teamName = args.options.teamName;
+    telemetryProps.channelId = args.options.channelId;
+    telemetryProps.channelName = args.options.channelName;
+    telemetryProps.tabId = args.options.tabId;
+    telemetryProps.tabName = args.options.tabName;
+    return telemetryProps;
+  }
+
+  public getTeamId(teamName: string): Promise<string> {
+    if (Utils.isValidGuid(teamName as string)) {
+      return Promise.resolve(teamName);
+    }
+
+    return new Promise<string>((resolve: (result: string) => void, reject: (error: any) => void): void => {
+      const teamRequestOptions: any = {
+        url: `${this.resource}/v1.0/me/joinedTeams?$filter=displayName eq '${encodeURIComponent(teamName)}'`,
+        headers: {
+          accept: 'application/json;odata.metadata=none'
+        },
+        json: true
+      }
+
+      request
+        .get<{ value: Team[] }>(teamRequestOptions)
+        .then((res: { value: Team[] }): Promise<string> => {
+          const teamItem: Team | undefined = res.value[0];
+
+          if (res.value.length > 1) {
+            Promise.reject(`Multiple Microsoft Teams team found with ids ${res.value.map(x => x.id)}`);
+          }
+
+          if (!teamItem) {
+            Promise.reject(`The specified team does not exist in the Microsoft Teams`);
+          }
+
+          const teamId: string = res.value[0].id;
+          return Promise.resolve(teamId);
+        })
+        .then((teamId: string): void => {
+          resolve(teamId);
+        })
+        .catch((error?: string): void => {
+          reject(error);
+        });
+    });
+  }
+
+  public getChannelId(teamId: string, channelName: string): Promise<string> {
+    if (Utils.isValidTeamsChannelId(channelName as string)) {
```

Comment:
This seems like unnecessary duplication of the validation logic which we could remove.